### PR TITLE
inhand interaction refactor

### PR DESCRIPTION
**Pull request by silicons**
Adds and standardizes hooks for

Activate inhand (wield)
Unique action (pump shotgun, etc)
Defensive toggle (shield modes)
Defensive trigger (shield special)

### CODE REVIEW

File: code/game/objects/items-interaction.dm [hunk 1]
```diff
@@ -0,0 +1,134 @@
...
+ *
+ * @params
+ * * actor - the event_args that spawned this call
+ * * user - The person using us in hand; stop using this, this is deprecated
+ *
+ * @return TRUE to signal to overrides to stop the chain and do nothing.
+ */
+/obj/item/proc/attack_self(mob/user, datum/event_args/actor/actor = new /datum/event_args/actor(user))
+	// SHOULD_NOT_OVERRIDE(TRUE)
+	// SHOULD_NOT_SLEEP(TRUE)
```
Comment (deathride58) [inline code/game/objects/items-interaction.dm:23]: why are these commented

**Review by deathride58 — COMMENTED** [review]
Just a tiny nitpick; looks good for testmerge

File: code/game/objects/items-interaction.dm (hunk 1 above)
Comment (TheLordME) [reply code/game/objects/items-interaction.dm:23]: comments are not for storing old code

**Review by TheLordME — CHANGES_REQUESTED** [review]
Comments are not for storing old code

**Pull request merged**